components/disclosure/profile-tabs.tsx

```diff
@@ -20,25 +21,26 @@ interface ProfileTabsProps {
   articles: Awaited<ReturnType<typeof getArticleList>>
 }
 
+function joinArticles(fetchedArticles: Awaited<ReturnType<typeof fetchArticlesByUsername>>, articles: Awaited<ReturnType<typeof getArticleList>>) {
+  return articles.map((article) => {
+    const isLiked = fetchedArticles.some(
+      (item) => item.articleURL === article.slug,
+    )
+    return isLiked ? article : null // 一致した場合は記事を返し、一致しない場合はnull
+  })
+    // nullを除外して最終的なlikedArticlesを得る
+    .filter((article) => article !== null)
+}
+
```

Comment:
`top-page-tabs.tsx`と同じロジックなので共通化してください

それと型定義や定数をこう書き換えるのはどうですか？
```suggestion
function joinArticles(fetchedArticles: Awaited<ReturnType<typeof fetchArticlesByUsername>> | Awaited<ReturnType<typeof fetchBookmarksByUsername>>, articles: Awaited<ReturnType<typeof getArticleList>>) {
  return articles.map((article) => {
    const isHit = fetchedArticles.some(
      (item) => item.articleURL === article.slug,
    )
    return isHit ? article : null // 一致した場合は記事を返し、一致しない場合はnull
  })
    // nullを除外して最終的なlikedArticlesを得る
    .filter((article) => article !== null)
}
```

Reply:
良いと思います！それで行きましょう